SwrveSDK/SDK/Talk/SwrveBaseCampaign.m

```diff
@@ -167,21 +168,24 @@ -(void)loadRulesFrom:(NSDictionary*)json
 -(void)loadTriggersFrom:(NSDictionary*)json
 {
     NSArray* jsonTriggers = [json objectForKey:@"triggers"];
+    
     if (!jsonTriggers) {
         DebugLog(@"Error loading triggers", nil);
         return;
     }
     
-    for (NSString* trigger in jsonTriggers){
+    for (NSDictionary* trigger in jsonTriggers){
         if (trigger) {
-            [self.triggers addObject:[trigger lowercaseString]];
+
+            SwrveTrigger *swrveTrigger = [[SwrveTrigger alloc] initWithDictionary:trigger];
+            [self.triggers addObject:swrveTrigger];
         }
     }
     
     DebugLog(@"Campaign Triggers:", nil);
-    for (NSString* trigger in self.triggers){
+    for (SwrveTrigger *trigger in self.triggers){
 #pragma unused(trigger)
-        DebugLog(@"- %@", trigger);
+        DebugLog(@"name:- %@", trigger.eventName);
```

Comment:
print the conditions also? Perhaps a toString method (or whatever is equivalent in obj c) in SwrveTrigger?